### PR TITLE
test: add a test to check for correct internal links, and fix them

### PR DESCRIPTION
**Pull request by pb8o**
Add a test to check that internal links in markdown files are correct.

Also restrict both markdown tests to only check for files within the git repo.

## Changes

- Restrict style checks to files commited in git. This helps with very messy workspaces.
- A new check for markdown internal links.
- A new devtool command to quickly run style checks.

## Reason

This should help internal and external contributors run style checks quickly, which is one of the DevEx pain points we currently have.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by sudanl0** [comment]
Tagging the issue which was created for this: https://github.com/firecracker-microvm/firecracker/issues/4342 
also a general question: why not use the mlc tool mentioned in the issue which can find internal and external broken links?

**Comment by pb8o** [comment]
> Tagging the issue which was created for this: #4342 also a general question: why not use the mlc tool mentioned in the issue which can find internal and external broken links?

Because I would rather not introduce a new dependency if it doesn't buy us much. Checking for internal links is pretty simple. And checking for external links I think it should not be done as part of CI (we can sync on that offline).

File: tests/integration_tests/style/test_markdown.py [hunk 1]
```diff
@@ -32,3 +33,18 @@ def test_markdown_style():
...
     ), "Some markdown files need formatting. Either run `./tools/devtool sh mdformat .` in the repository root, or apply the above diffs manually."
+
+
+def test_markdown_internal_links():
+    """Make sure markdown internal links work"""
+
+    md_files = [Path(x) for x in utils_repo.git_repo_files(root="..", glob="*.md")]
+    for md_file in md_files:
+        txt = md_file.read_text(encoding="utf-8")
+        for link in re.findall(r"\[.+?\]\((?P<link>.+?)\)", txt, re.DOTALL):
```
Comment (roypat) [inline tests/integration_tests/style/test_markdown.py:44]: I don't think we use footnotes for internal links very often, but if it's easy to do, maybe we could check those too?

File: tests/integration_tests/style/test_licenses.py [hunk 2]
```diff
@@ -90,16 +88,10 @@ def test_for_valid_licenses():
     """
     Test that all *.py, *.rs and *.sh files contain a valid license.
     """
-    python_files = utils.get_files_from(
-        find_path="..", pattern="*.py", exclude_names=EXCLUDE
-    )
```
Comment (roypat) [inline tests/integration_tests/style/test_licenses.py:95]: I think this eliminates all uses of `utils.get_files_from`, should we delete it?

**Review by roypat — APPROVED** [review]

File: tests/integration_tests/style/test_licenses.py (hunk 2 above)
Comment (pb8o) [reply tests/integration_tests/style/test_licenses.py:95]: Good point. Removing.

File: tests/integration_tests/style/test_markdown.py (hunk 1 above)
Comment (pb8o) [reply tests/integration_tests/style/test_markdown.py:44]: hmm yeah we only have 3 at the moment.

```
% git grep -c "\[\^\d+\]" *.md
CONTRIBUTING.md:2
SPECIFICATION.md:4
```

Comment (roypat) [reply tests/integration_tests/style/test_markdown.py:42]: Yeah, maybe not worth the effort

**Review by roypat — DISMISSED** [review]

**Review by roypat — APPROVED** [review]

**Review by sudanl0 — APPROVED** [review]

**Review by kalyazin — APPROVED** [review]

**Pull request merged**